### PR TITLE
Add optional log checking to RegressionTest.

### PR DESCRIPTION
**Pull request by jdcove2**
This PR adds optional log checking to RegressionTest. It is disabled by default and enabled by overriding getLogbackLoggerName() and returning the name of the logger to track.

### CODE REVIEW

**Review by arp-0984 — APPROVED** [review]
Looks reasonable to me

**Review by fbruton — CHANGES_REQUESTED** [review]
Recommend updating an existing regression test to exercise this new feature.

**Review by sambish5 — APPROVED** [review]

**Review by fbruton — APPROVED** [review]

File: src/test/java/emissary/test/core/junit5/RegressionTest.java [hunk 1]
```diff
@@ -175,6 +199,26 @@ protected void checkAnswersPreHook(final Document answers, final IBaseDataObject
         }
     }
 
+    /**
+     * This method returns the logger name to capture log events from or null if log events are not to be checked.
+     * 
+     * @return the logger name to capture log events from or null (the default) if log events are not to be checked.
+     */
+    protected String getLogbackLoggerName() {
```
Comment (drivenflywheel) [inline src/test/java/emissary/test/core/junit5/RegressionTest.java:207]: We don't _need_ to change this, but this is a clumsy design pattern.  Why override a method if the goal is just to set a String value?  RegressionTest should just read a config file entry and use it to assign a field value.  That pattern is deeply ingrained in the emissary code base. 
To be honest, we should be doing the same with respect to generating answer files.  Requiring a `generateAnswers()` override just to toggle a boolean value is poor design, and makes devs write throwaway code.

```ini
LOGGER_TO_CAPTURE = "name of logger"
GENERATE_ANSWER_FILES = "true"
```

Comment (drivenflywheel) [reply src/test/java/emissary/test/core/junit5/RegressionTest.java:207]: Fine, skip the config file, use protected fields in RegressionTest.  This is Object-Oriented Design, but awful encapsulation.

**Pull request merged**